pkg/client/introduce/client.go

```diff
@@ -9,40 +7,27 @@ SPDX-License-Identifier: Apache-2.0
 package introduce
 
 import (
-	"encoding/json"
 	"errors"
 	"fmt"
 
-	"github.com/google/uuid"
-
-	"github.com/hyperledger/aries-framework-go/pkg/common/log"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/didexchange"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/introduce"
-	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
-const invitationEnvelopePrefix = "invitation_envelope_"
-
-var logger = log.New("aries-framework/introduce/client")
-
 // Provider contains dependencies for the introduce protocol and is typically created by using aries.Context()
 type Provider interface {
 	Service(id string) (interface{}, error)
-	StorageProvider() storage.Provider
 }
 
 // Client enable access to introduce API
 type Client struct {
 	service.Event
-	service    service.DIDComm
-	store      storage.Store
-	defaultInv *didexchange.Invitation
-	newUUID    func() string
+	service service.DIDComm
 }
 
 // New return new instance of introduce client
-func New(ctx Provider, inv *didexchange.Invitation) (*Client, error) {
+func New(ctx Provider) (*Client, error) {
```

Comment:
how does the client async continue the protocol without event.Continue?

Reply:
Done! I have added the `UnfinishedActions` function